### PR TITLE
Add Rococo test network

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: runtime/rococo/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,1065 @@
...
+	pub const SlashDeferDuration: staking::EraIndex = 28;
+	pub const RewardCurve: &'static PiecewiseLinear<'static> = &REWARD_CURVE;
+	pub const MaxNominatorRewardedPerValidator: u32 = 64;
+	// quarter of the last session will be for election.
+	pub const ElectionLookahead: BlockNumber = EPOCH_DURATION_IN_BLOCKS / 4;
+	pub const MaxIterations: u32 = 10;
+	pub MinSolutionScoreBump: Perbill = Perbill::from_rational_approximation(5u32, 10_000);
+}
+
+impl staking::Trait for Runtime {
```
Comment (gavofyork) [inline runtime/rococo/src/lib.rs:303]: any need for staking? why not just a poa chain?

Comment (bkchr) [reply runtime/rococo/src/lib.rs:303]: Good point.

@cecton could you please remove the staking pallet and regenerate the chain specs?

Comment (cecton) [reply runtime/rococo/src/lib.rs:303]: To regenerate the chain specs I use build-spec --chain=rococo and I remove the entries in bootbodes in the JSON?

Comment (cecton) [reply runtime/rococo/src/lib.rs]: Nevermind it's fine I think. Can you double check?

**Comment by cecton** [comment]
We should merge if the reviewers are ok

**Comment by cecton** [comment]
I have no idea what cargo is doing but it's using a different substrate commit (8180062d - latest) than the one in the Cargo.lock (e824e8ab0)... this makes the update harder. Working on this now...

**Comment by cecton** [comment]
:rofl:

File: service/src/lib.rs [hunk 2]
```diff
@@ -804,6 +819,39 @@ pub fn westend_new_full(
 	Ok((service, client, handles))
 }
 
+/// Create a new Westend service for a full node.
```
Comment (rphmeier) [inline service/src/lib.rs:822]:
```suggestion
/// Create a new Rococo service for a full node.
```

ironically the same mistake as fixed above : )

File: runtime/rococo/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,1025 @@
...
+		max_piece_count: 40,
+		test_precision: 0_005_000,
+	);
+}
+
+parameter_types! {
+	// 28 eras for unbonding (7 days).
+	pub const BondingDuration: staking::EraIndex = 28;
+	// 28 eras in which slashes can be cancelled (7 days).
+	pub const SlashDeferDuration: staking::EraIndex = 28;
```
Comment (rphmeier) [inline runtime/rococo/src/lib.rs:292]:
```suggestion
	pub const SlashDeferDuration: staking::EraIndex = 27;
```

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**